test/unitary/session/streams/StreamStorage.cpp

```diff
@@ -0,0 +1,228 @@
+#include <c/core/session/stream/stream_storage.c>
+}
+
+#define BUFFER_SIZE           128
+#define HISTORY               4
+#define OFFSET                8
+
+class StreamStorageTest : public testing::Test
+{
+public:
+    StreamStorageTest()
+    {
+        uxr_init_stream_storage(&storage);
+        EXPECT_EQ(0, storage.input_best_effort_size);
+        EXPECT_EQ(0, storage.output_best_effort_size);
+        EXPECT_EQ(0, storage.input_reliable_size);
+        EXPECT_EQ(0, storage.output_reliable_size);
+
+    }
+
+    virtual ~StreamStorageTest()
+    {
+    }
+
+protected:
+    uxrStreamStorage storage;
+    uint8_t ob_buffer[BUFFER_SIZE / HISTORY];
+    uint8_t or_buffer[BUFFER_SIZE];
+    uint8_t ir_buffer[BUFFER_SIZE];
+
+    static void on_new_fragment(struct ucdrBuffer* ub, struct uxrOutputReliableStream* stream)
+    {
+        (void) ub;
+        (void) stream;
+    }
+
+    static FragmentationInfo on_get_fragmentation_info(uint8_t* buffer)
+    {
+        (void) buffer;
+        return NO_FRAGMENTED;
+    }
+
+public:
+    static int output_best_effort_reset_times;
+    static int input_best_effort_reset_times;
+    static int output_reliable_reset_times;
+    static int input_reliable_reset_times;
+
+    static bool output_best_effort_initialized;
+    static bool input_best_effort_initialized;
+    static bool output_reliable_initialized;
+    static bool input_reliable_initialized;
+
+    static bool output_reliable_up_to_date;
+};
+
+int StreamStorageTest::output_best_effort_reset_times = 0;
+int StreamStorageTest::input_best_effort_reset_times = 0;
+int StreamStorageTest::output_reliable_reset_times = 0;
+int StreamStorageTest::input_reliable_reset_times = 0;
+
+bool StreamStorageTest::output_best_effort_initialized = false;
+bool StreamStorageTest::input_best_effort_initialized = false;
+bool StreamStorageTest::output_reliable_initialized = false;
+bool StreamStorageTest::input_reliable_initialized = false;
+
+bool StreamStorageTest::output_reliable_up_to_date = false;
+TEST_F(StreamStorageTest, Reset)
+{
+    output_best_effort_reset_times = 0;
+    input_best_effort_reset_times = 0;
+    output_reliable_reset_times = 0;
+    input_reliable_reset_times = 0;
+
+    (void) uxr_add_input_best_effort_buffer(&storage);
+    (void) uxr_add_output_best_effort_buffer(&storage, ob_buffer, BUFFER_SIZE / HISTORY, OFFSET);
+    (void) uxr_add_input_reliable_buffer(&storage, ir_buffer, BUFFER_SIZE, HISTORY, on_get_fragmentation_info);
+    (void) uxr_add_output_reliable_buffer(&storage, or_buffer, BUFFER_SIZE, HISTORY, OFFSET, on_new_fragment);
+
+    uxr_reset_stream_storage(&storage);
+
+    EXPECT_EQ(UXR_CONFIG_MAX_OUTPUT_BEST_EFFORT_STREAMS, output_best_effort_reset_times);
+    EXPECT_EQ(UXR_CONFIG_MAX_INPUT_BEST_EFFORT_STREAMS, input_best_effort_reset_times);
+    EXPECT_EQ(UXR_CONFIG_MAX_OUTPUT_RELIABLE_STREAMS, output_reliable_reset_times);
+    EXPECT_EQ(UXR_CONFIG_MAX_INPUT_RELIABLE_STREAMS, input_reliable_reset_times);
```

Comment:
Internally, when you perform the reset function, this will call the reset in all streams of the stream manager. This check ensures that the internal input reliable stream is called the same times as the streams number.